### PR TITLE
Improve xlsx import

### PR DESCRIPTION
**Pull request by alex-Arc**
- switch from `node-xlsx` to `xlsx`
- evaluate excel formulars functions
- allow space and upper case in custom import

### CODE REVIEW

File: apps/server/src/api-data/excel/excel.service.ts [hunk 1]
```diff
@@ -23,32 +24,33 @@ export async function saveExcelFile(filePath: string) {
   if (extname(filePath) != '.xlsx') {
     throw new Error('Wrong file format');
   }
-  excelData = xlsx.parse(filePath, { cellDates: true });
+  excelData = xlsx.readFile(filePath, { cellDates: true, cellFormula: false });
```
Comment (cpvalente) [inline apps/server/src/api-data/excel/excel.service.ts:27]: question: did you try to see what happens if we pass `cellFormula: true`?

File: apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts [hunk 2]
```diff
@@ -82,3 +83,19 @@ export function getPersistedOptions(): NamedImportMap {
...
+        return ic.ontimeName === key;
+      })
+    ) {
+      // const ontimeName = key as unknown as string;
+      custom.push({ ontimeName: key, importName: label });
+    }
+  });
+
+  return importMap;
+}
```
Comment (cpvalente) [inline apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts:101]: question: could you please add some comments as to what this does?
We prefer explicit in favour of terser logic. Terser is a job for the builder

File: apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts [hunk 3]
```diff
@@ -82,3 +83,19 @@ export function getPersistedOptions(): NamedImportMap {
     return namedImportMap;
   }
 }
+
+export function addExistingCustomFields(importMap: NamedImportMap, customFields: CustomFields): NamedImportMap {
+  const custom = importMap.custom;
```
Comment (cpvalente) [inline apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts:88]: question: is this just a convenience rellabelling? 
did you check that we are mutating the original `importMap`?

File: apps/server/src/api-data/excel/excel.service.ts (hunk 1 above)
Comment (alex-Arc) [reply apps/server/src/api-data/excel/excel.service.ts:27]: yes then we get the raw excel formular

File: apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts [hunk 4]
```diff
@@ -82,3 +83,23 @@ export function getPersistedOptions(): NamedImportMap {
...
+        return customFieldFromImportMap.ontimeName === key;
+      })
+    ) {
+      // If the custom field dosn't exist in the import map we add it
+      newCustomFields.push({ ontimeName: key, importName: label });
+    }
+  });
+
+  return { ...importMap, custom: newCustomFields };
+}
```
Comment (cpvalente) [inline apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts:105]: I am not entirely confident on this logic here. I could be missing something.
Could be also nice to add a single happy path test just as documentation

`const customFieldsInImportMap = importMap.custom;`

This reassignemnt doesnt seem to do much for us, but could provide the indirection of us thinking that the object is different, I suggest we remove it

`const newCustomFields = new Array(...customFieldsInImportMap);`

I am not sure why do we need to call the Array constructor, is spreading into a new array not enough? `newCustomFields = [...customFieldsInImportMap];`

`return { ...importMap, custom: newCustomFields };`
why do we need to spread in the end? the `importMap` is given by the consumer and used as a reference here.
Would it make more sense for the consumer to get the `importMap` and pass only the `custom` part here?

**Comment by cpvalente** [comment]
@coderabbitai full review

**Review by cpvalente — APPROVED** [review]
Thank you alex, good work

**Pull request merged**